### PR TITLE
parsify: init at 2.0.1

### LINKED ISSUE #272310: Package request: parsify
**Project description**

"Next generation notepad-based calculator, built with extendibility and privacy in mind."

it's [Soulver](https://soulver.app/) but for platforms other than macos.

distributed as a deb and an appimage: https://github.com/parsify-dev/desktop/tree/main

**Metadata**

* homepage URL: https://parsify.app/
* source URL: n/a
* license: proprietary
* platforms: darwin, linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

closes https://github.com/NixOS/nixpkgs/issues/272310

https://github.com/parsify-dev/desktop/tree/v2.0.1
https://parsify.app/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 274845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>parsify</li>
  </ul>
</details>

File: pkgs/by-name/pa/parsify/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib,
+appimageTools,
+fetchurl
+}:
```
Comment (sersorrel) [inline pkgs/by-name/pa/parsify/package.nix:4]: nitpicky style comment, but someone else'll complain about it even if I don't:

```suggestion
{ lib
, appimageTools
, fetchurl
}:
```

**Review by keysmashes — APPROVED** [review]
Result of `nixpkgs-review pr 274845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>parsify</li>
  </ul>
</details>

File: pkgs/by-name/pa/parsify/package.nix (hunk 1 above)
Comment (kashw2) [reply pkgs/by-name/pa/parsify/package.nix]: fixed in https://github.com/NixOS/nixpkgs/pull/274845/commits/82038709e94c983608708239f2b8ec82ed8b2532

File: pkgs/by-name/pa/parsify/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  extraInstallCommands = ''
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
+  '';
+
+  meta = with lib; {
+    description = "Next generation notepad-based calculator, built with extendibility and privacy in mind.";
+    homepage = "https://parsify.app/";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ kashw2 ];
+    platforms = platforms.linux;
```
Comment (WolfangAukang) [inline pkgs/by-name/pa/parsify/package.nix:24]:
```suggestion
    platforms = platforms.linux;
    mainProgram = "parsify";
```

File: pkgs/by-name/pa/parsify/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    url = "https://github.com/parsify-dev/desktop/releases/download/v${version}/Parsify-${version}-linux-x86_64.AppImage";
+    hash = "sha256-ltWqRW+cBvuUJzhya62WsBY5zqIua9xhilxfd9gr24A=";
+  };
+
+  extraInstallCommands = ''
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
+  '';
+
+  meta = with lib; {
+    description = "Next generation notepad-based calculator, built with extendibility and privacy in mind.";
```
Comment (WolfangAukang) [inline pkgs/by-name/pa/parsify/package.nix:20]:
```suggestion
    description = "Next generation notepad-based calculator, built with extendibility and privacy in mind";
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]
As a suggestion, it would be cool to add the desktop file that the AppImage offers, mostly because this is an application with a GUI.

File: pkgs/by-name/pa/parsify/package.nix (hunk 2 above)
Comment (kashw2) [reply pkgs/by-name/pa/parsify/package.nix:33]: Thanks for the catch, amended in https://github.com/NixOS/nixpkgs/pull/274845/commits/b5162bfa84e012181547a8dfc0ff12b27a20aec9

File: pkgs/by-name/pa/parsify/package.nix (hunk 3 above)
Comment (kashw2) [reply pkgs/by-name/pa/parsify/package.nix]: https://github.com/NixOS/nixpkgs/pull/274845/commits/b5162bfa84e012181547a8dfc0ff12b27a20aec9

**Comment by kashw2** [comment]
Following the suggestion from @WolfangAukang I've added the .desktop file that the App Image provides and it's iconography.

**Review by WolfangAukang — APPROVED** [review]

**Pull request merged**